server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/CrossSpaceCompactionTask.java

```diff
@@ -79,11 +83,14 @@ public CrossSpaceCompactionTask(
     this.unseqTsFileResourceList = tsFileManager.getUnsequenceListByTimePartition(timePartition);
     this.performer = performer;
     this.hashCode = this.toString().hashCode();
+    this.memoryCost = memoryCost;
   }
 
   @Override
   protected void doCompaction() {
     try {
+      SystemInfo.getInstance().addCompactionMemoryCost(memoryCost);
```

Comment:
what if the allocateMemory throw interrupt exception without succeeding in adding?

Reply:
I move allocateMemory in a seperate try-catch block.